pkg/codegen/internal/test/testdata/aws-fargate.pp

```diff
@@ -0,0 +1,108 @@
+// Read the default VPC and public subnets, which we will use.
+vpc = invoke("aws:ec2:getVpc", {
+	default = true
+})
+subnets = invoke("aws:ec2:getSubnetIds", {
+	vpcId = vpc.id
+})
+
+// Create a security group that permits HTTP ingress and unrestricted egress.
+resource webSecurityGroup "aws:ec2:SecurityGroup" {
+	vpcId = vpc.id
+	egress = [{
+		protocol = "-1"
+		fromPort = 0
+		toPort = 0
+		cidrBlocks = ["0.0.0.0/0"]
+	}]
+	ingress = [{
+		protocol = "tcp"
+		fromPort = 80
+		toPort = 80
+		cidrBlocks = ["0.0.0.0/0"]
+	}]
+}
+
+// Create an ECS cluster to run a container-based service.
+resource cluster "aws:ecs:Cluster" {}
+
+// Create an IAM role that can be used by our service's task.
+resource taskExecRole "aws:iam:Role" {
+	assumeRolePolicy = {
```

Comment:
Is `toJSON` call missing here or is it not required?

Reply:
It's not required. The `assumeRolePolicy` property accepts arbitrary objects and serializes them transparently.

Reply:
@pgavlin That is true in Node.js, but we don't support that yet in other languages do we?  Do we handle code generation correctly for this in other languages?

Reply:
We don't generate code for this: it's implemented in the provider itself.

Reply:
It's just `string` in the schema, so I wonder how C# should handle this situation. Default to `toJSON` when anything is trying to assign to a string? Anyway, one more use case for type annotations, it seems.

Reply:
> It's just string in the schema

The input property has this type:

```
                "assumeRolePolicy": {
                    "oneOf": [
                        {
                            "type": "string"
                        },
                        {
                            "type": "string",
                            "$ref": "#/types/aws:iam/documents:PolicyDocument"
                        }
                    ],
                    "description": "The policy that grants an entity permission to assume the role.\n"
                },
```

Reply:
We don't codegen these union types of `string | JSON-object` in anything except Node.js - right?

Reply:
Ah, it's actually `string | PolicyDocument` in the schema, but C# doesn't support oneOf I guess.

Reply:
> We don't codegen these union types of string | JSON-object in anything except Node.js - right?

Yeah, I was typing that out. The lang-specific code generators will need to recognize that.

Reply:
(we don't even have anyway to codegen JSON object types in any othr language - https://github.com/pulumi/pulumi-terraform-bridge/issues/132)

Reply:
I *think* supporting this is literally just pulumi/pulumi-terraform-bridge#132.

Technically we might want to avoid code-generating example code that won't work in other languages till that is fixed.  But since this is used so rarely - might be better to just prioritize fixing 132.

Reply:
> (we don't even have anyway to codegen JSON object types in any othr language - pulumi/pulumi-terraform-bridge#132)

The Pulumi schema and associated code generators can do this. We don't have support for adding additional object types to the Pulumi schemas generated from TF schemas.